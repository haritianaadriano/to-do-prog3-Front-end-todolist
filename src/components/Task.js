import Checkbox from "./CheckBox"

export default function Task({todo, index, toDone}){
    return(
        <div>
            <li 
                key={index}
                data-testid="task-id"
            >
                {todo}
                <Checkbox
                    toDone={toDone}
                    index={index}
                />
            </li>
        </div>
    )
}