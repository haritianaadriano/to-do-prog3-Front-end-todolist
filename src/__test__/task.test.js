import {render, cleanup, screen} from "@testing-library/react";
import Task from "../components/Task";

describe("Task test", () => {
    const todoMock = "todo-1";

    it("Task appears in document", () => {
        const taskElement = screen.getByTestId("task-id");
        expect(taskElement).toBeInTheDocument();
    })

    it("todo mocking appears in task component", () => {
        const taskElement = render(<Task todo = {todoMock}/>);
        expect(taskElement).toHaveTextContent(todoMock);
    })
})