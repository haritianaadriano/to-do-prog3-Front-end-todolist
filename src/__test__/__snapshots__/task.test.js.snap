// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task test [snapshot] Task renders correctly 1`] = `
<div>
  <li
    data-testid="task-id"
  >
    todo-1
    <div>
      <input
        onChange={[Function]}
        type="checkbox"
      />
    </div>
  </li>
</div>
`;
